src/include/splash/basetypes/ColTypeString.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2013-2015 Felix Schmitt, Axel Huebl
+ *
+ * This file is part of libSplash.
+ *
+ * libSplash is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * libSplash is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libSplash.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COLTYPESTRING_H
+#define	COLTYPESTRING_H
+
+#include "splash/CollectionType.hpp"
+
+namespace splash
+{
+    /** Variable and fixed length strings
+     *
+     * Since C-strings are NULL-terminated, len must be +1
```

Comment:
Looks good to me in general, but I am a little worried that it might be unintuitive to use since you have to be very careful not to miss the additional byte for the NULL terminator with fixed-length strings.